File: deep-aerospike-native/src/main/java/com/stratio/deep/examples/java/WritingEntityToAerospikeNative.java
package com.stratio.deep.examples.java;

import com.stratio.deep.aerospike.direct.config.AerospikeConfigFactory;
import com.stratio.deep.aerospike.direct.config.AerospikeDeepJobConfig;
import com.stratio.deep.core.context.DeepSparkContext;
import com.stratio.deep.examples.utils.ContextProperties;
import com.stratio.deep.examples.utils.MessageTestEntity;
import org.apache.log4j.Logger;
import org.apache.spark.rdd.RDD;
import scala.Tuple2;

import java.util.List;

/**
 * Example of writing an Entity to Aerospike using the native extractor.
 */
public class WritingEntityToAerospikeNative {

    private static final Logger LOG = Logger.getLogger(WritingEntityToAerospikeNative.class);
    public static List<Tuple2<String, Integer>> results;

    public static void main(String[] args) {
        doMain(args);
    }

    public static void doMain(String[] args) {
        String job = "java:writingEntityToAerospikeNative";

        String host = "localhost";
        Integer port = 3000;

        String namespace = "test";
        String inputCollection = "message";
        String outputCollection = "messageOutput";

        // Creating the Deep Context where args are Spark Master and Job Name
        ContextProperties p = new ContextProperties(args);
        DeepSparkContext deepContext = new DeepSparkContext(p.getCluster(), job, p.getSparkHome(),
                p.getJars());

        AerospikeDeepJobConfig<MessageTestEntity> inputConfigEntity =
                AerospikeConfigFactory.createAerospike(MessageTestEntity.class).host(host).port(port).namespace(namespace)
                        .set(inputCollection);

        RDD<MessageTestEntity> inputRDDEntity = deepContext.createRDD(inputConfigEntity);

        AerospikeDeepJobConfig<MessageTestEntity> outputConfigEntity =
                AerospikeConfigFactory.createAerospike(MessageTestEntity.class).host(host).port(port).namespace(namespace)
                        .set(outputCollection);

        deepContext.saveRDD(inputRDDEntity, outputConfigEntity);

        deepContext.stop();
    }
}

File: deep-aerospike-native/src/main/java/com/stratio/deep/examples/scala/WritingEntityToAerospikeNative.scala
/*
 * Copyright 2014 Aerospike, Inc.
 *
 * Portions may be licensed to Aerospike, Inc. under one or more
 * contributor license agreements.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you
 * may not use this file except in compliance with the License. You
 * may obtain a copy of the License at
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
 * implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */
package com.stratio.deep.examples.scala

import com.stratio.deep.aerospike.direct.config.{AerospikeConfigFactory, AerospikeDeepJobConfig}
import com.stratio.deep.aerospike.direct.extractor.AerospikeNativeEntityExtractor
import com.stratio.deep.core.context.DeepSparkContext
import com.stratio.deep.examples.utils.{MessageTestEntity, ContextProperties}
import org.apache.spark.rdd.RDD

/**
 * Example of writing an Entity to Aerospike using the native extractor.
 */
object WritingEntityToAerospikeNative {

  def main(args: Array[String]) {
    doMain(args)
  }

  def doMain(args: Array[String]) {
    val job: String = "scala:writingEntityToAerospikeNative"
    val host: String = "localhost"
    val port: Integer = 3000
    val namespace: String = "test"
    val inputSet: String = "message"
    val outputSet: String = "messageOutput"

    val p: ContextProperties = new ContextProperties(args)

    val deepContext: DeepSparkContext = new DeepSparkContext(p.getCluster, job, p.getSparkHome, p.getJars)

    val inputConfigEntity: AerospikeDeepJobConfig[MessageTestEntity] = AerospikeConfigFactory.createAerospike(classOf[MessageTestEntity]).host(host).port(port).namespace(namespace).set(inputSet)
    inputConfigEntity.setExtractorImplClass(classOf[AerospikeNativeEntityExtractor[MessageTestEntity]])
    inputConfigEntity.initialize

    val inputRDDEntity: RDD[MessageTestEntity] = deepContext.createJavaRDD(inputConfigEntity)

    val outputConfigEntity: AerospikeDeepJobConfig[MessageTestEntity] = AerospikeConfigFactory.createAerospike(classOf[MessageTestEntity]).host(host).port(port).namespace(namespace).set(outputSet)
    outputConfigEntity.setExtractorImplClass(classOf[AerospikeNativeEntityExtractor[MessageTestEntity]])
    outputConfigEntity.initialize

    DeepSparkContext.saveRDD(inputRDDEntity, outputConfigEntity)

    deepContext.stop

  }

}
